### PR TITLE
cluster-autoscaler: initial commit

### PR DESCRIPTION
**Pull request by mhrabovcin**
JIRA: https://jira.d2iq.com/browse/D2IQ-65258

### CODE REVIEW

File: stable/cluster-autoscaler/templates/clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    resources:
+      - events
+      - endpoints
+    verbs:
+      - create
+      - patch
+  - apiGroups:
+    - ""
+    resources:
+    - pods/eviction
```
Comment (hectorj2f) [inline stable/cluster-autoscaler/templates/clusterrole.yaml]: we are gonna need to perform actions on the konvoycluster objects

File: stable/cluster-autoscaler/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    {{- if .Values.affinity }}
+      affinity:
+{{ toYaml .Values.affinity | indent 8 }}
+    {{- end }}
+    {{- if .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.nodeSelector | indent 8 }}
+    {{- end }}
+      serviceAccountName: {{ template "serviceaccount.name" . }}
+      tolerations:
```
Comment (hectorj2f) [inline stable/cluster-autoscaler/templates/deployment.yaml]: we want it to be deployed on the control-plane by default

File: stable/cluster-autoscaler/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+description: Scales worker nodes within autoscaling groups.
+icon: https://github.com/kubernetes/kubernetes/blob/master/logo/logo.png
+name: cluster-autoscaler
+version: 7.1.0
+appVersion: 1.17.1
+home: https://github.com/kubernetes/autoscaler
+sources:
+- https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler
+- https://github.com/spotinst/kubernetes-autoscaler/tree/master/cluster-autoscaler
+maintainers:
```
Comment (shaneutt) [inline stable/cluster-autoscaler/Chart.yaml]: Are these upstream maintainers?

If so, this is a `stable/` chart so we need to add our own maintainers so as to not confuse our end-users.

Comment (mhrabovcin) [reply stable/cluster-autoscaler/Chart.yaml]: Yes, these are upstream maintainers @shaneutt . Should I replace completely maintainers section or just extend it with our names?

Comment (hectorj2f) [reply stable/cluster-autoscaler/Chart.yaml]: @shaneutt we are gonna apply changes to this vanilla chart, so I'd say we could add them as part of a different PR, imho.

Comment (shaneutt) [reply stable/cluster-autoscaler/Chart.yaml]: > Yes, these are upstream maintainers @shaneutt . Should I replace completely maintainers section or just extend it with our names?

For now, replace it with our names. We should not claim those people are responsible for our fork.

**Pull request merged**